Plugins~/Src/MeshSyncClientBlender/msblenGeometryNodesUtils.cpp

```diff
@@ -33,21 +35,34 @@ namespace blender {
             to_mat4x4(rotation) *
             to_mat4x4(rotation180) *
             scale44(scale_z);
+        
+        m_camera_light_correction = to_mat4x4(rotate_x(90 * DegToRad));
     }
 
     /// <summary>
     /// Converts the world matrix from blender to Unity coordinate systems
     /// </summary>
     /// <param name="blenderMatrix"></param>
     /// <returns></returns>
-    float4x4 GeometryNodesUtils::blenderToUnityWorldMatrix(const float4x4& blenderMatrix) {            
+    float4x4 GeometryNodesUtils::blenderToUnityWorldMatrix(const Object* obj, const float4x4& blenderMatrix) {
+        
+        float4x4 result = blenderMatrix;
 
-        return 
-            m_blender_to_unity_world *
-            blenderMatrix *
+        if (msblenUtils::is_camera(obj) || msblenUtils::is_light(obj)) {
+             msblenEntityHandler::applyCorrectionIfNeeded(obj, result);
+        }
+
+        result = m_blender_to_unity_world *
+            result *
             m_blender_to_unity_local;
-    }
+        
+        if (msblenUtils::is_camera(obj) || msblenUtils::is_light(obj)) {
```

Comment:
It seems that wee are doing the correction twice:
One before applying the world matrix, and another one after.
Does it have to be like this, and is it possible to combine them ?